Python/Product/Analysis/Interpreter/InterpreterArchitecture.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Reflection;
+
+namespace Microsoft.PythonTools.Interpreter {
+    public struct InterpreterArchitecture : 
+        IFormattable,
+        IEquatable<InterpreterArchitecture>,
+        IComparable<InterpreterArchitecture>
+    {
+        public ProcessorArchitecture Value { get; set; }
+
+        public InterpreterArchitecture(ProcessorArchitecture arch) {
```

Comment:
The flexible ToString(), mainly. Makes it possible to do string.Format("{0}{1: ()}", name, arch) to get "2.7" or "2.7 (32-bit)"

---
In reply to: [75406677](https://github.com/Microsoft/PTVS/pull/1536#discussion_r75406677) [](ancestors = 75406677)